R/dplyr.R

```diff
@@ -51,11 +51,7 @@ bind_rows <- function(x, y = NULL) {
       y[[nme]] <- NA
     }
   }
-  rbind.data.frame(x, y)
-}
-
-if_else <- function(condition, true, false, missing = NULL) {
```

Comment:
I think we might just keep `if_else()`. In dplyr it's a stricter variant of `ifelse()`, it's just not currently implemented like this but could be.

Reply:
I just found it confusing how the function is defined now, because at first look, people may think it's the strict version from dplyr but it's not.

Reply:
The dplyr implementation seems to have too many dependencies (in terms of internal functions used) to be dragged over here. But something like:

```r
stopifnot(length(condition) == length(true))
stopifnot(length(condition) == length(false))
if (!is.null(missing)) stop("missing arg not yet implemented")
```

might just do the job for now.

Reply:
Ok, the problem is that we deliberately allow for an output that has length zero, e.g. in `compute_indent_indices()`, so we have used `ifelse()` all the time.Should we allow for length zero response in "our" implementation of `if_else()`?

Reply:
Perhaps:

```r
stopifnot(length(true) %in% c(length(condition), 1L))
stopifnot(length(false) %in% c(length(condition), 1L))
if (!is.null(missing)) stop("missing arg not yet implemented")
```

?

Reply:
Well, if length zero should be supported, I guess your `1L` should be a `0L`?

Reply:
We support zero length for the condition only.

Reply:
We have
```r
if_else(is.null(token_closing),
  npd,
  which(pd$token %in% token_closing) - 1
)
```
So `false` is of length zero if `token` is not in `token_closing`

Reply:
And the condition is of length one.

Reply:
Or should we simply use 
```r
if (cond) {
  ret <- integer(0)
} else {
  ret <- xyz(fun)
}
```
For those cases?

Reply:
``` r
dplyr::if_else(TRUE, 1L, integer(0))
#> Error: `false` must be length 1 (length of `condition`), not 0
```

`if_else()` is a safety wrapper to ensure that the output length matches the input length. I guess we'll need something different, perhaps a regular `if()` for other cases.

Reply:
(Just the way you suggested.)

Reply:
Ok, will do so.